lib/hooks/api/useFetchMetricStats.ts

```diff
@@ -0,0 +1,41 @@
+import useSWR, { Fetcher } from "swr";
+import { publicApiFetcher } from "lib/utils/public-api-fetcher";
+
+type UseFetchMetricStatsProps = {
+  repository: string;
+  variant: "stars" | "forks"; // TODO: add other MetricCard types
+  range: number;
+};
+
+export type StatsType = {
+  bucket: string;
+  star_count?: number;
+  forks_count?: number;
+};
+
+export function useFetchMetricStats({ repository, variant, range }: UseFetchMetricStatsProps) {
+  const query = new URLSearchParams();
+  query.set("repo", repository);
+  query.set("range", range.toString());
+
+  const endpoint = () => {
+    switch (variant) {
+      case "stars":
+        return `histogram/stars?${query.toString()}`;
+      case "forks":
+        return `histogram/forks?${query.toString()}`;
```

Comment:
refactor (blocking): The template string will convert the `query` object to a string like `query.toString()`.

```suggestion
        return `histogram/stars?${query}`;
      case "forks":
        return `histogram/forks?${query}`;
```